frb/turb_scattering.py

```diff
@@ -0,0 +1,228 @@
+""" Codes for Scattering by ISM
+  Based on formalism derived by Macquart & Koay 2013, ApJ, 776, 125
+  Modified by Prochaska & Neeleman 2017
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import numpy as np
+import pdb
+
+from scipy.special import gamma
+
+from astropy import constants as const
+from astropy import units as u
+
+# Constants
+const_re = const.alpha**2 * const.a0
+
+
+class Turbulence(object):
+    """ Class for turbulence calculations in a plasma
+    Primarily used for scattering calculations
+    """
+    def __init__(self, ne, l0, L0, zL=0., beta=11./3, SM=None, verbose=True):
+        """
+        Parameters
+        ----------
+        ne : Quantity, optional
+          Electron density 
+        l0 : Quantity
+          Inner scale
+        L0 : Quantity, optional
+          Outer scale
+        SM : Quantity, optional
+          Generally calculated but can be input
+        zL : float, optional
```

Comment:
why does z_L get treated differently? Isn't this just as much a property as L0, lo, and ne of the turbulence object. In other words, not setting it does not give any errors, yet it is likely not giving the correct value (unless the object you consider is at z=0).

Reply:
right, it simply defaults to 0